mobile-app/app/screens/AppNavigator/screens/Dex/CompositeSwap/CompositeSwapScreen.tsx

```diff
@@ -221,32 +222,48 @@ export function CompositeSwapScreen ({ route }: Props): JSX.Element {
       return
     }
 
-    setIsTokenSelectDisabled(true)
+    const tokenSelectOption: DexParamList['CompositeSwapScreen']['tokenSelectOption'] = route.params.tokenSelectOption ?? {
+      from: {
+        isDisabled: true,
+        isPreselected: true
+      },
+      to: {
+        isDisabled: true,
+        isPreselected: true
+      }
+    }
+
+    setIsFromTokenSelectDisabled(tokenSelectOption.from.isDisabled)
+    setIsToTokenSelectDisabled(tokenSelectOption.to.isDisabled)
 
     const pair = pairs.find((pair) => pair.data.id === route.params.pair?.id)
     if (pair !== undefined) {
-      onTokenSelect({
-        tokenId: pair.data.tokenA.id,
-        available: new BigNumber(pair.data.tokenA.reserve),
-        token: {
-          displaySymbol: pair.data.tokenA.displaySymbol,
-          symbol: pair.data.tokenA.symbol,
-          name: '' // not available in API
-        },
-        reserve: pair.data.tokenA.reserve
-      }, 'FROM')
-      onTokenSelect({
-        tokenId: pair.data.tokenB.id,
-        available: new BigNumber(pair.data.tokenB.reserve),
-        token: {
-          displaySymbol: pair.data.tokenB.displaySymbol,
-          symbol: pair.data.tokenB.symbol,
-          name: '' // not available in API
-        },
-        reserve: pair.data.tokenB.reserve
-      }, 'TO')
+      if (tokenSelectOption.from.isPreselected) {
```

Comment:
Can we do an early return here at line 240? To avoid nested if-block.
e.g.: `if (pair !== undefined) return`

Reply:
Agree. Upddated